crates/starknet-server/src/api/http/endpoints/mint_token.rs

```diff
@@ -46,20 +50,27 @@ pub(crate) async fn mint(
     Json(request): Json<MintTokensRequest>,
     Extension(state): Extension<HttpApiHandler>,
 ) -> HttpApiResult<Json<MintTokensResponse>> {
-    // increase balance
     let mut starknet = state.api.starknet.write().await;
+    let erc20_address = match request.unit {
+        Some(FeeUnits::WEI) | None => {
+            ContractAddress::new(Felt::from_prefixed_hex_str(ETH_ERC20_CONTRACT_ADDRESS).unwrap())
+                .unwrap()
+        }
+        Some(FeeUnits::FRI) => {
+            ContractAddress::new(Felt::from_prefixed_hex_str(STRK_ERC20_CONTRACT_ADDRESS).unwrap())
+                .unwrap()
+        }
+    };
+    let unit = if let Some(unit) = request.unit { unit } else { FeeUnits::WEI };
```

Comment:
This unnecessary logic is introduced again.
This should be done using `.unwrap_or` before `match`, and then apply `match` on the unwrapped object.

Reply:
Right somehow I missed that, mainly because I didn't know that it would behave like this in case of `None` 

Fixed in e1a993eb0b1dcdd95373ec7a776ac8242b952388